File: tiq-e/others/1/main.py
from time import perf_counter


class Solution:
    def hammingWeight(self, n: int) -> int:
        num_of_1s = 0
        for i in range(64):
            num_of_1s += (n & 1)
            n = n >> 1
        return num_of_1s


n = 3

start = perf_counter()
result = Solution().hammingWeight(n)

end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/others/1/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn hammingWeight(n: u32) -> i32 {
        n.count_ones() as i32
    }
}

fn main() {
    let n: u32 = 3;

    let start = Instant::now();
    let result = Solution::hammingWeight(n);

    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
